src/extensions/default/AutoUpdate/main.js

```diff
@@ -421,15 +378,35 @@ define(function (require, exports, module) {
      * @param {Array} updates - array object containing info about updates
      */
     function _updateProcessHandler(updates) {
-        //If no checksum field is present then we're setting it to 0, just as a safety check,
-        // although ideally this situation should never occur in releases post its introduction.
-        var platformInfo = getPlatformInfo(),
-            buildName = updates[0].versionString,
-            checksum  = (updates[0].checksums) ? updates[0].checksums[platformInfo.OS] : 0;
-
-        var updateParams = getDownloadInfo(buildName, platformInfo.extension);
-        updateParams.latestBuildNumber = updates[0].buildNumber;
-        updateParams.checksum = checksum;
+
+        var platform = getPlatformInfo(),
+            checksum,
+            downloadURL,
+            installerName,
+            assetInfo;
+
+        assetInfo = updates[0].assetInfo;
```

Comment:
Retain the `updates[0]` in a local var for once upfront.

Reply:
@swmitra , addressed in 44fef7e